### PR TITLE
time between states metric

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by rudigiesler** [comment]
@hodgestar: Ready for review

File: lib/metrics-helper.js [hunk 1]
```diff
@@ -110,6 +110,28 @@ var MetricsHelper = Eventable.extend(function(self, im) {
...
+        var metadata = user.metadata;
+
+        if (typeof metadata[label] === 'undefined'){
+            metadata[label] = 0;
+        }
+
+        user.metadata[label] += 1;
+    };
+
+    self._get_metadata = function(user, label) {
```
Comment (hodgestar) [inline lib/metrics-helper.js]: Surprised that `_get_metadata` clears the value. Wondering whether we shouldn't perhaps call this `_reset_metadata` instead (couldn't think of a great suggestion, but at least `reset` makes it obvious that the value is being cleared?

File: lib/metrics-helper.js [hunk 2]
```diff
@@ -110,6 +110,28 @@ var MetricsHelper = Eventable.extend(function(self, im) {
...
+    // Gets the specified metadata value, resets to zero, and returns value
+        var metadata_value = user.metadata[label];
```
Comment (hodgestar) [inline lib/metrics-helper.js:132]: We should handle the case where `user.metadata[label]` is `undefined` (probably this shouldn't happen, but might occur, if for example, someone adds a metric after the initial deploy and users are mid-way through a menu).

**Comment by hodgestar** [comment]
Otherwise looks good.

**Comment by rudigiesler** [comment]
@hodgestar Ready for review again

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**